app/ts/Subproblem/SubproblemButtons/AddSubproblemButton/AddSubproblemUtil.test.ts

```diff
@@ -0,0 +1,619 @@
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          }
+        ],
+        distributions: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'empty'
+          }
+        ],
+        relativePerformances: []
+      } as IWorkspace;
+
+      const warnings = getMissingValueWarnings(
+        dataSourceInclusions,
+        alternativeInclusions,
+        workspace
+      );
+
+      const expectedWarnings: string[] = [noSmaaWarning];
+      expect(warnings).toEqual(expectedWarnings);
+    });
+
+    it('should warn about missing deterministic values when SMAA values are present', () => {
+      const dataSourceInclusions = {ds1Id: true};
+      const alternativeInclusions = {alt1Id: true};
+
+      const workspace = {
+        distributions: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          }
+        ],
+        effects: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'empty'
+          }
+        ],
+        relativePerformances: []
+      } as IWorkspace;
+
+      const warnings = getMissingValueWarnings(
+        dataSourceInclusions,
+        alternativeInclusions,
+        workspace
+      );
+
+      const expectedWarnings: string[] = [noDeterministicWarning];
+      expect(warnings).toEqual(expectedWarnings);
+    });
+
+    it('should warn about missing deterministic and SMAA values, when the problem has a cell with neither', () => {
+      const dataSourceInclusions = {ds1Id: true, ds2Id: true};
+      const alternativeInclusions = {alt1Id: true};
+
+      const workspace = {
+        distributions: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          },
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds2Id',
+            criterionId: 'crit1Id',
+            type: 'empty'
+          }
+        ],
+        effects: [
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds1Id',
+            criterionId: 'crit1Id',
+            type: 'empty'
+          },
+          {
+            alternativeId: 'alt1Id',
+            dataSourceId: 'ds2Id',
+            criterionId: 'crit1Id',
+            type: 'value',
+            value: 0.1
+          }
+        ],
+        relativePerformances: []
+      } as IWorkspace;
+
+      const warnings = getMissingValueWarnings(
+        dataSourceInclusions,
+        alternativeInclusions,
+        workspace
+      );
+
+      const expectedWarnings: string[] = [
+        noDeterministicWarning,
+        noSmaaWarning
+      ];
```

Comment:
This is actually a case that should never happen as it is not possible to input this during manual input. It can only happen when someone uploads an incorrect problem.